### ISSUE TITLE
About Token accuracy and Training

### ISSUE THREAD

**Issue by CIntellifusion** [valeoai/Halton-MaskGIT#26]
Thanks for you excellent work and cleaned code! 

I just want to double check that the training method in this repo is following original maskgit,right? 

Second, I am reproducing maskgit on other datasets, but the token accuracy is very low like 0.02 after 20 epoch on imagenet with 4k codebook size. 

Could provide some reference accuracy number on imagenet with 1k codebook size? Thanks for your kindly help. 

Also I am using a FSQ quantizer for imagenet, which may also cause issues. 

Thanks for your insights.

**Comment by llvictorll** [comment]
Hello @CIntellifusion,

I’m glad to hear my work has been helpful!

The training setup is quite similar to MaskGIT, but since the official repo didn’t release training code, it’s hard to say. For instance, we use classifier-free guidance (CFG), which isn’t mentioned in the original paper. You can refer to [this paper](https://arxiv.org/abs/2310.14400) for more details, but the performance match when using the same tokenizer, see [previous commit](https://github.com/valeoai/Halton-MaskGIT/tree/v1.0). AFAIK, this repository is the closer to the original work when using the confidence scheduler.

At 20 epochs, the model hasn’t converged yet, but for a codebook of 16k, low accuracy (~7%) is expected, though it doesn’t necessarily reflect generation quality. I’m not sure about the 1k codebook setup, but generally, smaller codebooks tend to yield higher accuracy.

Unfortunately, I don’t have experience with FSQ :/ 

Best,
